File: src/main/java/com/goudourasv/data/ratings/RatingEntity.java
package com.goudourasv.data.ratings;

import com.goudourasv.data.courses.CourseEntity;
import com.goudourasv.data.users.UserEntity;

import javax.persistence.*;
import java.util.UUID;

@Entity(name = "Rating")
public class RatingEntity {
    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    private CourseEntity courseEntity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private UserEntity userEntity;

    @Column
    private String message;

    @Column(name = "user_rating")
    private int userRating;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public CourseEntity getCourseEntity() {
        return courseEntity;
    }

    public void setCourseEntity(CourseEntity courseEntity) {
        this.courseEntity = courseEntity;
    }

    public UserEntity getUserEntity() {
        return userEntity;
    }

    public void setUserEntity(UserEntity userEntity) {
        this.userEntity = userEntity;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getUserRating() {
        return userRating;
    }

    public void setUserRating(int userRating) {
        this.userRating = userRating;
    }
}

File: src/main/java/com/goudourasv/data/ratings/RatingEntity.kt
package com.goudourasv.data.ratings

import com.goudourasv.data.courses.CourseEntity
import com.goudourasv.data.users.UserEntity
import java.util.*
import javax.persistence.*

@Entity(name = "Rating")
class RatingEntity(
    @Id
    @GeneratedValue
    var id: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    var courseEntity: CourseEntity,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var userEntity: UserEntity,

    @Column
    var message: String,

    @Column(name = "user_rating")
    var userRating: Int,
)
